### DISCUSSION TITLE
Anyone can do user scriptlets for Bromite browser?

### DISCUSSION THREAD

**Discussion by gsbnlda** [xarantolus/filtrite#13] [Ideas]
Hello, YouTube ads are blocked by scriptlets. None of the filters and user scriptlets blocking YT ads in bromite browser. anyone here /developer can make perfect like uBO for bromite. I really like bromite browser because it is faster than mull ,fennec, iceraven,firefox nightly.   These user scriptlets not perfect ; - https://github.com/xarantolus/bromite-userscripts,

https://greasyfork.org/scripts/436535-adguard-script-block-youtube-ads/code/AdGuard%20script%20block%20YouTube%20ads.user.js,

filters :- https://github.com/uBlockOrigin/uAssets/blob/dcbc568c0712108fc6617789bb72995e39cbcb1e/filters/filters.txt#L131,

https://raw.githubusercontent.com/uBlockOrigin/uAssets/master/filters/annoyances.txt,

May be we take adguard filters and make user.js file ? Thanks.

Posts :- https://www.reddit.com/r/androidapps/comments/usy5k6/youtube_ads_in_bromite_browser/ ,

https://www.reddit.com/r/uBlockOrigin/comments/usbddt/what_is_that_filter_block_ads_in_youtube/,

https://www.reddit.com/r/fossdroid/comments/uc6h89/filter_list_for_bromite/,

**Comment by xarantolus** [comment]
I kind of tried integrating scriptlets into the [cosmetic filter script](https://github.com/xarantolus/bromite-userscripts/tree/main/generate/cosmetic) before, but wasn't really sure if I got it right and then abandoned it, so yea currently these rules are not available in the script.

Basically one would have to [build this](https://github.com/AdguardTeam/Scriptlets) and add it at the top of the cosmetic script, then parse scriptlet rules (in the generator) and transform them into something that actually calls these functions in the sccript. It shouldn't be too hard, maybe I'll look into it again soon<sup>TM</sup>. I think there's also some syntax differences between uBO and AdGuard, but not sure

**Comment by xarantolus** [comment]
I added support for AdGuard scriptlets and it appears to work OK (I think; maybe the timing of scriptlet injections should be different).

You can try it out and tell me if youtube ads are blocked for you if you install [this cosmetic.user.txt](https://github.com/xarantolus/filtrite/files/8765602/cosmetic.user.txt) file :)

Edit: the filter you linked [here](https://github.com/uBlockOrigin/uAssets/blob/dcbc568c0712108fc6617789bb72995e39cbcb1e/filters/filters.txt#L131) is currently not supported, but there script contains an equivalent filter from another source

Another edit: Sometimes YouTube pages crash with the "Oh snap!" message, so something is wrong with the way snippets are currently handled.
